### PR TITLE
Mutate only status when calling UpdateStatus with fake client

### LINKED ISSUE #107319: client-go fake: UpdateStatus mutates spec
### What happened?

Calling UpdateStatus with fake client-go mutates spec

### What did you expect to happen?

UpdateStatus only mutates status, like with real client

### How can we reproduce it (as minimally and precisely as possible)?

```
package gateway

import (
	"context"
	"testing"

	corev1 "k8s.io/api/core/v1"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	"k8s.io/client-go/kubernetes/fake"
)

func TestUpdateStatus(t *testing.T) {
	f := fake.NewSimpleClientset()
	f.CoreV1().Pods("default").Create(
		context.Background(),
		&corev1.Pod{
			ObjectMeta: metav1.ObjectMeta{Name: "test-pod"},
			Spec:       corev1.PodSpec{NodeName: "test-node"},
		},
		metav1.CreateOptions{},
	)
	pod, _ := f.CoreV1().Pods("default").Get(context.Background(), "test-pod", metav1.GetOptions{})
	t.Log(pod.Spec.NodeName)
	f.CoreV1().Pods("default").UpdateStatus(
		context.Background(),
		&corev1.Pod{
			ObjectMeta: metav1.ObjectMeta{Name: "test-pod"},
		},
		metav1.UpdateOptions{},
	)
	pod, _ = f.CoreV1().Pods("default").Get(context.Background(), "test-pod", metav1.GetOptions{})
	t.Log(pod.Spec.NodeName)
}
```

Result:
```
=== RUN   TestUpdateStatus
    kube_fake_test.go:23: test-node
    kube_fake_test.go:32: 
--- PASS: TestUpdateStatus (0.00s)
```

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

k8s.io/client-go v0.22.4

</details>

### Cloud provider

N/A

### OS version

_No response_

### Install tools

<details>

</details>

### Container runtime (CRI) and and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by haoruan**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Fake client only update the status when calling the UpdateStatus, will not touch anything like spec on existing obj.
#### Which issue(s) this PR fixes:

Fixes #107319

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/client-go/testing/fixture.go [hunk 1]
```diff
@@ -118,11 +118,35 @@ func ObjectReaction(tracker ObjectTracker) ReactionFunc {
 			if err != nil {
 				return true, nil, err
 			}
-			err = tracker.Update(gvr, action.GetObject(), ns)
+
+			var obj runtime.Object
+			switch action.GetSubresource() {
+			case "status":
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/testing/fixture.go:124]: Unfortunately, not all resources uniformly prevent writes to everything outside `Status` via the status subresource. Several resources allow modifying metadata (labels, annotations, finalizers, etc)

**Comment by haoruan** [comment]
Just updated with a simple, general way to prevent changes to spec, at least all resources are not allowed to set spec. I'm not sure if we'd like to fully implement or reuse the server side status update strategy, because there're some resources have specific fields preserved, like pods don't allow to modify DeletionTimestamp and OwnerReferences.

**Pull request closed without merging**